docs: add instructions about applying rulesets (#30)
guardian/gha-scala-library-release-workflow: docs/configuration.md

## Repo settings

* Disable [branch protection **rules**](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches)
  on any branch the workflow will be pushing to (ie the default branch). Note that PR #26 means that you can use branch protection rulesets to protect your default branch, so long as you allow your GitHub App to bypass those restrictions.
### Guardian developers 
* Set the custom property `production_status` to `production` to apply  branch protection to the default branch via a ruleset (this allows the Scala release app to bypass branch protection).

> * Create a separate 'Status checks' ruleset in your repo with the Branch protection property 'Require status checks to pass' -> 'Require branches to be up to date before merging' and add your repo's status check(s) to the 'Status checks that are required' list. For example, if your repo has a CI workflow with the name 'CI', then the setting would look like this:

Reviewer: > add your repo's status check(s)

I don't love the term "status checks" here. The status check is the thing being edited, not the thing being referenced. I think this sentence is really referring to their *workflow(s)*?

![status_checks.png](status_checks.png)

> * Also add the Scala release app to the ruleset bypass list as in the branch protection ruleset.

Reviewer: Do we need the word "Also" here?

Is this enough information for someone not super familiar with the topic to find and select the correct app?

## Repo settings

* Disable [branch protection **rules**](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches)

>   on any branch the workflow will be pushing to (ie the default branch). Note that PR #26 means that you can use branch protection rulesets to protect your default branch, so long as you allow your GitHub App to bypass those restrictions.

Reviewer 2: ```suggestion
  on any branch the workflow will be pushing to (ie the default branch). Note that PR #26 means that you can use rulesets to protect your default branch, so long as you allow your GitHub App to bypass those restrictions.
```

Also, as this specific library relies on a specific GH App, could we name it explicitly?

### Guardian developers 

> * Re-enable branch protections:

Reviewer 2: I think calling this re-enabling is potentially confusing, as we're enabling a slightly different thing.
```suggestion
* Enable branch protection rulesets:
```
might be a bit clearer?

  * Set the custom property `production_status` to `production` to apply  branch protection to the default branch via a ruleset (this allows the Scala release app to bypass branch protection).

>   * Create a separate 'Status checks' ruleset in your repo with the Branch protection property 'Require status checks to pass' -> 'Require branches to be up to date before merging' and add your repo's workflow name(s) to the 'Status checks that are required' list. For example, if your repo has a CI workflow with the name 'CI', then the setting would look like this:

Reviewer 2: Someone unfamiliar with rulesets will probably assume a 'Status checks' ruleset means something specific, rather than the arbitrary(-ish) name we've assigned to the ruleset.

```suggestion
  * Create a second ruleset in your repo with the Branch protection property 'Require status checks to pass' -> 'Require branches to be up to date before merging'. Type your repo's workflow name(s) (not to be confused with the filename) into the 'Status checks that are required' box. For example, if your repo has a CI workflow with the name 'CI', then the setting would look like this:
```
The first sentence is a bit long so maybe we could break it up a bit? I've suggested a couple of other bits that I think would help with clarity

>   ![status_checks.png](status_checks.png)

Reviewer 2: Putting in this diagram was a good call!

>   * Add the Scala release app to the ruleset bypass list (see the 'Branch protection' ruleset for an example).

Reviewer 2: Duplicate of an earlier comment, could we reference the app by name either here or above?

Author: I wasn't sure if referencing the app in a public repo was a good idea - so deliberately kept it vague - and Guardian users will see the name of the app in the ruleset created by adding the custom property. WDYT @NovemberTang ?
